components/concurrency_manager/src/lock_table.rs

```diff
@@ -299,4 +305,19 @@ mod test {
         lock_table.for_each(|h| collect(h, &mut found_locks));
         assert_eq!(found_locks, expect_locks);
     }
+
+    #[tokio::test]
+    async fn test_lock_key_when_handle_exists() {
+        let lock_table: LockTable = LockTable::default();
+        let key = Key::from_raw(b"key");
+        let guard = lock_table.lock_key(&key).await;
+        let handle = lock_table.get(&key).unwrap();
+        drop(guard);
+
+        let guard2 = lock_table.lock_key(&key).await;
+        drop(handle);
+
+        let handle = lock_table.get(&key).unwrap();
+        assert_eq!(Arc::as_ptr(&handle), Arc::as_ptr(guard2.handle()));
+    }
```

Comment:
How about add some more lines dropping both the `handle` and `guard2` and try `lock_key` again, and check the pointer?